src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/prisons/SupportedPrisonsRepository.kt

```diff
@@ -0,0 +1,40 @@
+package uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.prisons
+
+import org.hibernate.Hibernate
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.stereotype.Repository
+import java.time.Instant
+import java.time.format.DateTimeFormatter
+import javax.persistence.Entity
+import javax.persistence.Id
+import javax.persistence.Table
+
+@Repository
+interface SupportedPrisonsRepository : JpaRepository<SupportedPrison, String> {
+  fun findByActive(active: Boolean): List<SupportedPrison>
+}
+
+@Entity
+@Table(name = "supported_prisons")
+data class SupportedPrison(
+  @Id
+  val code: String = "",
+  val active: Boolean,
+  val updatedBy: String,
+  val updated: Instant,
+) {
+  override fun equals(other: Any?): Boolean {
```

Comment:
Just wondering, why override  hasCode and equals for these data classes?

Reply:
It's to do with Hibernate and storing entities in collections - basically if you update an entity then add it to a collection it should replace the existing entity in that collection, not add another element. There's a good article about it [here](https://vladmihalcea.com/the-best-way-to-implement-equals-hashcode-and-tostring-with-jpa-and-hibernate/).

Intellij issues a warning for JPA entities if you don't override equals and hashcode... I use a plugin called JPA Buddy to generate the equals and hascode for me.